### PR TITLE
Added option to disable webpage preview in message history. Fix #1216

### LINKED ISSUE #1216: Disable/option to disable link preview
As i also posted [here](https://github.com/telegramdesktop/tdesktop/issues/687#issuecomment-237016009) i would really like to see the linkpreview disabled, or a checkbox or something to disable it myself.

The link preview feature is really annoying me. I use Telegram (webclient) frequently, and the auto preview of URL's is making it harder for my chat partner to read the conversation in the layout as i meant it.
Please create an option to disable the link preview.

### PR DESCRIPTION
**Pull request by foysalit**
This is my first time contributing to telegram and I am quite new to Open Source contribution. Please feel free to correct me or suggest any changes required.

### CODE REVIEW

File: app/js/directives.js [hunk 1]
```diff
@@ -432,28 +432,38 @@ angular.module('myApp.directives', ['myApp.filters'])
...
-        $scope.openPhoto = AppPhotosManager.openPhoto
-        $scope.openEmbed = function ($event) {
-          if ($scope.media.webpage &&
-            $scope.media.webpage.embed_url) {
-            AppWebPagesManager.openEmbed($scope.media.webpage.id, $scope.messageId)
-            return cancelEvent($event)
-          }
-        }
+        Storage.get('disable_webpage_preview').then(function (previewDisabled) {
+          $scope.previewDisabled = !!previewDisabled;
```
Comment (auchri) [inline app/js/directives.js]: `!!previewDisabled == previewDisabled` 😉

Comment (foysalit) [inline app/js/directives.js]: yes you're right but when the app is first run, the storage value is not going to exist and that should give some weird value i.e: `undefined` or `null` depending on how the `Storage` service is built. doing `!!` should cover for that case too and give a solid `Boolean` type. Unless the storage service already handles it? I didn't look under the hood of that service.

**Comment by dhowe** [comment]
status?

**Comment by zhukov** [comment]
Hi there. Nice PR, but I would change it to something more generic. Setting like "hide all medias in chat history"  would be useful. And when turning it on, it would replace all the media contents (including webpage previews) with links, like "[Photo]", "[Web Preview]" or smth like that. Click on that link would show the attachment.

File: app/js/directives.js (hunk 1 above)
Comment (Ryuno-Ki) [inline app/js/directives.js]: @auchri Not quite. `!!previewDisabled` casts `previewDisabled` to a Boolean by first negating it (thereby casting it to a Boolean) and then invert the value (that is, falsy values become `false` and truthy `true`).

**Comment by zhukov** [comment]
Closing for now. Feel free to open an issue and discuss this.

**Pull request closed without merging**